tests/test_deserialization.py

```diff
@@ -826,6 +826,32 @@ class AliasingUserSerializer(Schema):
         assert result['email'] == 'foo@bar.com'
         assert result['age'] == 42
 
+    def test_deserialize_with_attribute_param_error_returns_field_name_not_attribute_name(self):
+        class AliasingUserSerializer(Schema):
+            username = fields.Email(attribute='email')
+            years = fields.Integer(attribute='age')
+        data = {
+            'username': 'foobar.com',
+            'years': '42'
+        }
+        result, errors = AliasingUserSerializer().load(data)
+        assert errors
+        assert errors['username'] == [u'"foobar.com" is not a valid email address.']
+
+    def test_deserialize_with_attribute_param_error_returns_load_from_not_attribute_name(self):
+        class AliasingUserSerializer(Schema):
+            name = fields.String(load_from='Name')
+            username = fields.Email(attribute='email', load_from='UserName')
+            years = fields.Integer(attribute='age', load_from='Years')
+        data = {
+            'Name': 'Mick',
+            'UserName': 'foobar.com',
+            'years': 'abc'
+        }
+        result, errors = AliasingUserSerializer().load(data)
+        assert errors['UserName'] == [u'"foobar.com" is not a valid email address.']
```

Comment:
What I did was just follow the behavior of where the data is actually
loaded from.  I thought it would be easier for the client to introspect if
we handed back the name of the field they specified instead of the one
specified by load_from.  `load_from` only gets used as fallback and I
thought I would reflect that in the error handling.

On Tue, Jul 7, 2015 at 5:12 AM, Steven Loria <notifications@github.com>
wrote:

> In tests/test_deserialization.py
> <https://github.com/marshmallow-code/marshmallow/pull/232#discussion_r34032640>
> :
>
> > +        result, errors = AliasingUserSerializer().load(data)
> > +        assert errors
> > +        assert errors['username'] == [u'"foobar.com" is not a valid email address.']
> > +
> > +    def test_deserialize_with_attribute_param_error_returns_load_from_not_attribute_name(self):
> > +        class AliasingUserSerializer(Schema):
> > +            name = fields.String(load_from='Name')
> > +            username = fields.Email(attribute='email', load_from='UserName')
> > +            years = fields.Integer(attribute='age', load_from='Years')
> > +        data = {
> > +            'Name': 'Mick',
> > +            'UserName': 'foobar.com',
> > +            'years': 'abc'
> > +        }
> > +        result, errors = AliasingUserSerializer().load(data)
> > +        assert errors['UserName'] == [u'"foobar.com" is not a valid email address.']
>
> OK, I see now why you'd want the error keys to be the same as load_from
> because the field names could be considered internal aliases for the value
> passed by the client. However, I'm still not clear on why you would only
> want to use load_from when the value is missing from the input. It seems
> more consistent to always use load_from if it is set.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/marshmallow-code/marshmallow/pull/232/files#r34032640>
> .
>